File: website/adminscripts/export.js
#!/usr/bin/node

if (process.argv.length < 3) {
    console.log("Usage: export.js PATH_TO_DICTIONARY.sqlite")
    process.exit(1);
}

response = {
    setHeader: function() {},
    write: function(str) {process.stdout.write(str)},
    end: function() {}
}

const ops = require("../ops");
const sqlite3 = require('sqlite3');
const path = require("path");

var db_path = process.argv[2];
var dictID = path.basename(db_path, ".sqlite");
var db = new sqlite3.Database(db_path, sqlite3.OPEN_READONLY, function (err) {
    if (err) {
        console.log("Failed to open database at " + db_path);
        console.log(err);
        process.exit(2);
    }
});
ops.download(db, dictID, response);

File: website/adminscripts/export.py
#!/usr/bin/python3

import os.path
import sqlite3
import sys

if len(sys.argv) != 2:
    print("Usage: ./adminscripts/export.py PATH_TO_DICTIONARY.sqlite")
    sys.exit()

db_path = sys.argv[1]
dictID = os.path.basename(db_path).replace(".sqlite", "")
try:
    db = sqlite3.connect(db_path)
    db.row_factory = sqlite3.Row
    resxml = "<"+dictID+">\n"
    c = db.execute("select id, xml from entries")
    for r in c.fetchall():
        resxml += r["xml"]
        resxml += "\n"
    resxml += "</"+dictID+">"
    db.close()
    print(resxml)
except sqlite3.Error as e:
    print("Database error: %s" % e)
